File: TheepotAndroid/app/src/main/kotlin/nl/bsodetheepot/mobile/data/models/Kasboek.kt
package nl.bsodetheepot.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class KasboekType {
    @SerialName("inkomst") INKOMST,
    @SerialName("uitgave") UITGAVE,
}

/** Tabel `kasboek_entries`. */
@Serializable
data class KasboekEntry(
    val id: String,
    val periode: String,
    val categorie: String? = null,
    val datum: String? = null,
    val omschrijving: String? = null,
    val bedrag: Double,
    val type: KasboekType,
    @SerialName("aangemaakt_door") val aangemaaktDoor: String? = null,
    @SerialName("aangemaakt_op") val aangemaaktOp: String,
    val locatie: String,
    @SerialName("bonnetje_pad") val bonnetjePad: String? = null,
)

/** Tabel `kasboek_periode_status`: publicatiestatus per locatie/maand voor directie. */
@Serializable
data class KasboekPeriodeStatus(
    @SerialName("locatie_naam") val locatieNaam: String,
    val periode: String,
    val gepubliceerd: Boolean,
)

object KasboekCategorieen {
    val standaard = listOf("Omzet", "Inkopen", "Personeelskosten", "Overige kosten", "Materialen", "Huisvestingskosten")
}

File: TheepotMobile/TheepotMobile/Models/Kasboek.swift
import Foundation

enum KasboekType: String, Codable, CaseIterable {
    case inkomst, uitgave
}

/// Tabel `kasboek_entries`.
struct KasboekEntry: Codable, Identifiable {
    let id: String
    var periode: String
    var categorie: String?
    var datum: String?
    var omschrijving: String?
    var bedrag: Double
    var type: KasboekType
    var aangemaaktDoor: String?
    let aangemaaktOp: String
    var locatie: String
    var bonnetjePad: String?

    enum CodingKeys: String, CodingKey {
        case id, periode, categorie, datum, omschrijving, bedrag, type
        case aangemaaktDoor = "aangemaakt_door"
        case aangemaaktOp = "aangemaakt_op"
        case locatie
        case bonnetjePad = "bonnetje_pad"
    }
}

/// Tabel `kasboek_periode_status`: houdt bij of het kasboek van een locatie/maand
/// al gepubliceerd is voor directie (die het pas dan te zien krijgt).
struct KasboekPeriodeStatus: Codable {
    var locatieNaam: String
    var periode: String
    var gepubliceerd: Bool

    enum CodingKeys: String, CodingKey {
        case locatieNaam = "locatie_naam"
        case periode, gepubliceerd
    }
}

enum KasboekCategorieen {
    static let standaard = ["Omzet", "Inkopen", "Personeelskosten", "Overige kosten", "Materialen", "Huisvestingskosten"]
}
